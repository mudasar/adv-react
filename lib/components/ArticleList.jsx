import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { withStyles } from 'material-ui/styles';
import Paper from 'material-ui/Paper';
import Typography from 'material-ui/Typography';
import Article from './Article.jsx';

const styles = theme => ({
  root: theme.mixins.gutters({
    paddingTop: 16,
    paddingBottom: 16,
    marginTop: theme.spacing.unit * 3,
  }),
});



export class ArticleList extends Component {
  static propTypes = {

  }

  render() {

    const { classes, articles, articleActions } = this.props;

    return (
      <div>
          {Object.values(articles).map( (article) => <Article key={article.id} article={article} actions={articleActions} />)}
      </div>
    )
  }
}

export default withStyles(styles)(ArticleList);
