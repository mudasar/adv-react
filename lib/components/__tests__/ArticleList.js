import React from 'react';
import ArticleList from '../ArticleList.jsx';
import renderer from 'react-test-renderer';

describe('ArticleList', () => {

    const testProps = {
        articles:{
            a: {id: 'a'},
            b: {id: 'b'}
        },
        articleActions:{
            lookupAuthor: jest.fn(()=> ({}))
        }
    };

  it('renders correctly', () => {
    const tree = renderer.create(
        <ArticleList {...testProps} />
    ).toJSON();

   // console.log(tree.children[0].children);
    expect(tree.children.length).toBe(2);
    expect(tree).toMatchSnapshot();
  });
});
