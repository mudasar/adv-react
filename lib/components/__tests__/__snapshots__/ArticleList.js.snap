// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    className="MuiPaper-root-6 MuiPaper-shadow2-10 MuiPaper-rounded-7 Article-card-2"
  >
    <div
      className="MuiCardContent-root-33"
    >
      <p
        className="MuiTypography-root-34 MuiTypography-body1-43 Article-title-4"
      >
        Invalid Date
      </p>
      <h2
        className="MuiTypography-root-34 MuiTypography-headline-39"
      />
      <a
        href={undefined}
      >
        <span
          className="MuiTypography-root-34 MuiTypography-body1-43 Article-pos-5"
        >
           
        </span>
      </a>
      <p
        className="MuiTypography-root-34 MuiTypography-body1-43"
      />
    </div>
    <div
      className="MuiCardActions-root-58"
    >
      <button
        className="MuiButtonBase-root-75 MuiButton-root-60 MuiButton-sizeSmall-72 MuiCardActions-action-59"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-61"
        >
          Learn More
        </span>
        <span
          className="MuiTouchRipple-root-77"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiPaper-root-6 MuiPaper-shadow2-10 MuiPaper-rounded-7 Article-card-2"
  >
    <div
      className="MuiCardContent-root-33"
    >
      <p
        className="MuiTypography-root-34 MuiTypography-body1-43 Article-title-4"
      >
        Invalid Date
      </p>
      <h2
        className="MuiTypography-root-34 MuiTypography-headline-39"
      />
      <a
        href={undefined}
      >
        <span
          className="MuiTypography-root-34 MuiTypography-body1-43 Article-pos-5"
        >
           
        </span>
      </a>
      <p
        className="MuiTypography-root-34 MuiTypography-body1-43"
      />
    </div>
    <div
      className="MuiCardActions-root-58"
    >
      <button
        className="MuiButtonBase-root-75 MuiButton-root-60 MuiButton-sizeSmall-72 MuiCardActions-action-59"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-61"
        >
          Learn More
        </span>
        <span
          className="MuiTouchRipple-root-77"
        />
      </button>
    </div>
  </div>
</div>
`;
